store/store.go

```diff
@@ -493,35 +538,29 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string, devmode bool, aut
 	case resp.StatusCode == 404:
 		return nil, ErrSnapNotFound
 	case resp.StatusCode != 200:
-		tpl := "Ubuntu CPI service returned unexpected HTTP status code %d while looking for snap %q in channel %q"
+		// TODO: do this for all requests
+		tpl := "Ubuntu CPI service returned unexpected HTTP status code %d while getting details for snap %q in channel %q"
 		if oops := resp.Header.Get("X-Oops-Id"); oops != "" {
 			tpl += " [%s]"
 			return nil, fmt.Errorf(tpl, resp.StatusCode, name, channel, oops)
 		}
 		return nil, fmt.Errorf(tpl, resp.StatusCode, name, channel)
 	}
 
+	s.checkStoreResponse(resp)
+
 	// and decode json
-	var searchData searchResults
+	var remote snapDetails
 	dec := json.NewDecoder(resp.Body)
-	if err := dec.Decode(&searchData); err != nil {
+	if err := dec.Decode(&remote); err != nil {
 		return nil, err
 	}
 
-	switch len(searchData.Payload.Packages) {
-	case 0:
-		return nil, ErrSnapNotFound
-	case 1:
-		// whee
-	default:
-		logger.Noticef("expected at most one exact match search result for %q in %q channel, got %d.", name, channel, len(searchData.Payload.Packages))
-		return nil, fmt.Errorf("unexpected multiple store results for an exact match search for %q in %q channel", name, channel)
+	info := infoFromRemote(remote)
+	if !devmode && info.NeedsDevMode() {
```

Comment:
let me try to clarify what bothers me, most flags we pass to methods here do filtering, this one does a bit more though, we would need to mention this in the doc comment but then that's not a generallya  good sign,

about the many places, one should actually call ListRefresh not Snap,  and they can be probably reduced to one (for the time being) with some refactoring, function nesting